### DISCUSSION TITLE
Use individual config files for Routes, Database Schema, Wasp Config, Actions/Queries

### DISCUSSION THREAD

**Discussion by nwhitmont** [wasp-lang/wasp#1924] [Archived]
As a Wasp developer, it would be nice to see Routes/Pages, Prisma Schema, and Wasp app config separated into individual dedicated config files instead of one giant file.  This way the project could use a normal `schema.prisma` file, and the Wasp app config and Routes/Pages could be a normal JS/TS file that exports JSON config objects.

example file structure:
```
config/
  app.ts
  routes.ts
  schema.prisma
```

example `app.ts`:
```
export const appConfig = {
  wasp: {
    version: "^0.13.0"
  },
  title: "example-app",
  auth: {
    userEntity: User,
    methods: {
      usernameAndPassword: {}, // This is a very naive implementation, use 'email' in production instead
      //google: {}, // https://wasp-lang.dev/docs/integrations/google
      //gitHub: {}, // https://wasp-lang.dev/docs/integrations/github
      //email: {} // https://wasp-lang.dev/docs/guides/email-auth
    },
    onAuthFailedRedirectTo: "/login",
  }
}
```

example `routes.ts`:
```
import { HomePage, AdminPage } from "@components/pages"

export const routes = [
  {
    path: "/",
    authRequired: true,
    component: HomePage
  },
  {
    path: "/admin",
    authRequired: true,
    component: AdminPage
  },
]
```

etc.

**Comment by schenktobi** [comment]
I would like this as well since my .wasp files get really big sometimes and i would love to put e.g. my actions or route definitions into another file.

**Comment by Martinsos** [comment]
Thanks all, this is something we plan to have in the near future for sure! We don't have it yet because we want to do it properly, but we will certainly be implementing it!

**Comment by aslauris** [comment]
For someone using VS Code adding folding support to VS extension could be really helpful. Adding this config to `language-configuration.json` should do the trick: https://github.com/Microsoft/vscode/blob/main/extensions/javascript/javascript-language-configuration.json#L103

**Reply by Martinsos** [reply]
Thanks and I saw you made a PR, will be checking that out soon!
